dango/dex/src/core/xyk.rs

```diff
@@ -0,0 +1,195 @@
+use {
+    grug::{
+        Bounded, CoinPair, IsZero, MathResult, MultiplyFraction, MultiplyRatio, Number,
+        NumberConst, StdResult, Udec128, Uint128, ZeroExclusiveOneExclusive,
+    },
+    std::{cmp, iter},
+};
+
+pub fn add_initial_liquidity(
+    mut reserve: CoinPair,
+    deposit: CoinPair,
+) -> anyhow::Result<(CoinPair, Uint128)> {
```

Comment:
One thing I don't like about this PR is that this function isn't pure any more...